src/test/java/org/example/controllers/ReaderControllerTest.java

```diff
@@ -0,0 +1,174 @@
+
+  @Test
+  void shouldReturnNotFoundWhenReaderNotPresent() throws Exception {
+    var idTen = 10L;
+    when(readerService.findById(idTen)).thenReturn(Optional.empty());
+    mvc.perform(get(REQUEST_PATH + "/" + idTen)).andExpect(status().isNotFound());
+  }
+
+  @Test
+  void shouldReturnReaderWithId1() throws Exception {
+    var idOne = 1L;
+    when(readerService.findById(idOne)).thenReturn(Optional.of(getFistReader()));
+    var mvcResult = mvc.perform(get(REQUEST_PATH + "/" + idOne)).andExpect(status().isOk())
+        .andReturn();
+
+    var contentAsString = mvcResult.getResponse().getContentAsString();
+    var readerFromResponse = objectMapper.readValue(contentAsString, Reader.class);
+
+    assertAll(() -> assertEquals(idOne, readerFromResponse.getId()),
+        () -> assertEquals("reader", readerFromResponse.getName()));
+
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"Patric", "John Random", "Mike Douglas"})
+  void shouldSaveNewReader(String name) throws Exception {
+    var readerDTO = new NewReaderDTO(name);
+    var reader = new Reader(name);
+    reader.setId(5L);
+    when(readerService.addNewReader(any(Reader.class))).thenReturn(reader);
+
+    MvcResult mvcResult = mvc.perform(post(REQUEST_PATH).contentType("application/json")
+            .content(objectMapper.writeValueAsString(readerDTO))).andExpect(status().is(201))
+        .andReturn();
+
+    var content = mvcResult.getResponse().getContentAsString();
+    var readerFromResponse = objectMapper.readValue(content, Reader.class);
+
+    assertAll(() -> assertEquals(readerFromResponse.getId(), reader.getId()),
+        () -> assertEquals(name, readerFromResponse.getName()));
+
+  }
+
+  @ParameterizedTest
+  @DisplayName("When send reader with invalid field should return 400 and error messages")
+  @CsvSource({"!aut", "1", "$$"})
+  void shouldReturnBadRequestWhenReaderNotValid(String name) throws Exception {
+    var readerDTO = new NewReaderDTO(name);
+    var result = mvc.perform(post(REQUEST_PATH).contentType("application/json")
+            .content(objectMapper.writeValueAsString(readerDTO))).andExpect(status().isBadRequest())
+        .andReturn();
+
+    var now = LocalDateTime.now().minusMinutes(1L);
+    var expectedErrorResponseDTO = getResponseForInvalidFieldsNewReaderDTO(name);
+    var errorResponseDTO = getErrorResponseFromMvcResult(result, objectMapper);
+
+    assertAll(
+        () -> assertEquals(2, errorResponseDTO.errors().size(), "Expected two errors in fields"),
+        () -> assertTrue(errorResponseDTO.errors().containsAll(expectedErrorResponseDTO.errors()),
+            "Some errors in fields miss"),
+        () -> assertTrue(
+            errorResponseDTO.errorMessage().contentEquals(expectedErrorResponseDTO.errorMessage()),
+            "Error message is wrong"),
+        () -> assertTrue(
+            now.isBefore(LocalDateTime.parse(errorResponseDTO.date(), dateTimeFormatter)),
+            "Time is missed"));
```

Comment:
Assertion message is misleading. It's more like "Error response time must be not null and not in past".

Or you can just keep you message and just assertNotNull for this field. Also fine.